apps/drec-ui/src/apps/device-group/view/pages/UnreservedPage/UnreservedPage.effects.ts

```diff
@@ -0,0 +1,108 @@
+import { DeviceGroupModalsActionsEnum, useDeviceGroupModalsDispatch } from '../../context';
+import {
+    UnreservedFormFormValues,
+    useUnreservedDeviceGroupsTableLogic,
+    useUnreservedFilterFormLogic
+} from '../../../logic';
+import { useMediaQuery, useTheme } from '@material-ui/core';
+import { GenericFormProps } from '@energyweb/origin-ui-core';
+import { useAllDeviceFuelTypes, useUnreservedDeviceGroups } from '../../../data';
+import { useEffect, useState } from 'react';
+import { UnreservedDeviceGroupDTO } from '@energyweb/origin-drec-api-client';
+import { UseFormReset } from 'react-hook-form';
+
+export const useUnreservedPageEffects = () => {
+    const initialFormValues: UnreservedFormFormValues = {
+        country: [],
+        fuelCode: [],
+        installationConfiguration: '',
+        offTaker: '',
+        sector: '',
+        standardCompliance: '',
+        gridInterconnection: '',
+        commissioningDateRange: '',
+        capacityRange: ''
+    };
+    const [filterUnreserved, setFilterUnreserved] = useState(initialFormValues);
+    const { deviceGroups, isLoading: IsLoadingDeviceGroups } =
+        useUnreservedDeviceGroups(filterUnreserved);
+    const [selectedDeviceGroupList, setSelectedDeviceGroupList] = useState(deviceGroups);
+    const dispatchModals = useDeviceGroupModalsDispatch();
+
+    const { allTypes, isLoading: isDeviceTypesLoading } = useAllDeviceFuelTypes();
+
+    const { fields, initialValues, validationSchema, buttonText } = useUnreservedFilterFormLogic(
```

Comment:
you are not extracting `secondaryButtons` here and later not including them in `formData` at line 69.
So to avoid this better not destructure the returned object, but rather have it as follows: `const formConfig = ...`
And at line 69 `...formConfig`

Reply:
But anyway I don`t think I need the secondary button anymore. So I can remove it.

Reply:
Then also remove the commented `resetForm` from `useUnreservedFilterFormLogic` args

Reply:
Yep

Reply:
Done